src/game_api/script/usertypes/level_lua.hpp

```diff
@@ -1,13 +1,16 @@
 #pragma once
 
+#include "level_api_types.hpp"
+
 #include <string>
 #include <vector>
 
 #include <sol/forward.hpp>
 
-using ROOM_TEMPLATE = uint16_t;     // NoAlias
-using PROCEDURAL_CHANCE = uint32_t; // NoAlias
-using ROOM_TEMPLATE_TYPE = int32_t; // NoAlias
+using LAYER = int; // NoAlias
+constexpr LAYER LAYER_FRONT = 0;
```

Comment:
I think this should be uint8_t not LAYER, as LAYER was solely made to be used with enum_to_layer function, first, then you check for the actual layer with value it returns (except LAYER_BOTH). Ideally i was thinking to make it throw warning/error if you try to compare LAYER with int or any other variable that is not LAYER

Reply:
It's just a workaround for not having a LAYER enum on the c++ side. Used to compare values coming in from the API so there's no value in narrowing the type. Ideally we should have the enum on the C++ side

Reply:
i will add that in my pr, then we can fix which ever pr comes first